### PR TITLE
[chore] add govulncheck

### PR DESCRIPTION
**Pull request by sakshi-1505**
fixes https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4413

### CODE REVIEW

**Comment by sakshi-1505** [comment]
@codeboten @pellared Please take a look

File: Makefile [hunk 1]
```diff
@@ -77,7 +77,10 @@ $(GORELEASE): PACKAGE=golang.org/x/exp/cmd/gorelease
 GOJSONSCHEMA = $(TOOLS)/go-jsonschema
 $(GOJSONSCHEMA): PACKAGE=github.com/atombender/go-jsonschema
 
-tools: $(GOLANGCI_LINT) $(MISSPELL) $(GOCOVMERGE) $(STRINGER) $(PORTO) $(MULTIMOD) $(DBOTCONF) $(CROSSLINK) $(GOTMPL) $(GORELEASE) $(GOJSONSCHEMA)
+GOVULNCHECK = $(TOOLS)/govulncheck
+ $(TOOLS)/govulncheck: PACKAGE=golang.org/x/vuln/cmd/govulncheck
```
Comment (pellared) [inline Makefile:81]:
```suggestion
$(TOOLS)/govulncheck: PACKAGE=golang.org/x/vuln/cmd/govulncheck
```

Comment (pellared) [inline Makefile:81]:
```suggestion
$(GOVULNCHECK): PACKAGE=golang.org/x/vuln/cmd/govulncheck
```

File: Makefile [hunk 2]
```diff
@@ -149,12 +152,20 @@ go-mod-tidy/%:
 misspell: | $(MISSPELL)
 	@$(MISSPELL) -w $(ALL_DOCS)
 
+.PHONY: govulncheck
+govulncheck: $(ALL_GO_MOD_DIRS:%=govulncheck/%)
+govulncheck/%: DIR=$*
+govulncheck/%:
```
Comment (pellared) [inline Makefile:158]:
```suggestion
govulncheck/%: | $(GOVULNCHECK)
```

**Review by pellared — APPROVED** [review]
I fixes some little issues. Looks good. We just need to do something like here https://github.com/open-telemetry/opentelemetry-go/pull/4607 to fix the reported issues.

Thanks for your contribution 👍

**Review by hanyuancheung — APPROVED** [review]

**Comment by MrAlias** [comment]
Please fix the issues this uncovers so the CI system will pass and this can be merged.

**Comment by pellared** [comment]
@sakshi-1505 Thanks 🎉

**Pull request merged**